pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -680,9 +685,15 @@ async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
                 self.provider_response_id = chunk.response_id
 
             raw_finish_reason = candidate.finish_reason
+            provider_details_dict: dict[str, Any] = {}
             if raw_finish_reason:
-                self.provider_details = {'finish_reason': raw_finish_reason.value}
+                provider_details_dict['finish_reason'] = raw_finish_reason.value
                 self.finish_reason = _FINISH_REASON_MAP.get(raw_finish_reason)
+            if self._provider_timestamp is not None:
+                # _provider_timestamp is always None in Google streaming cassettes
+                provider_details_dict['timestamp'] = self._provider_timestamp  # pragma: no cover
```

Comment:
Same as what I mentioned on the last review: this will now be called every single time in the loop, even if it was already set. 

So maybe we just set `self.provider_details = {'timestamp': self._provider_timestamp}` _before the loop_, and in here only store `finish_reason`?